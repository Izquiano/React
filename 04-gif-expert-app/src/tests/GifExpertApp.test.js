import React from 'react'
import GiftExpertApp from '../GifExpertApp'
import { shallow } from "enzyme";


describe('Pruebas sobre el componente <GiftExpertApp />', () => {
  test('debe mostrar el componente de manera correcta ', () => {
    const wrapper = shallow(<GiftExpertApp  />);
    expect(wrapper).toMatchSnapshot();
  })
  test('Debe dde mostrar una lista de categorias', () => {

    const categories = ['One punch', 'Dragon ball']
    const wrapper = shallow(<GiftExpertApp defaultCategories={ categories } />);

    expect( wrapper ).toMatchSnapshot()
    expect( wrapper.find('GifGrid').length ).toBe( categories.length )

    
  })
  
  
})




