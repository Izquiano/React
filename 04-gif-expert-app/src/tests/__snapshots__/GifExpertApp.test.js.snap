// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GiftExpertApp /> Debe dde mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre el componente <GiftExpertApp /> debe mostrar el componente de manera correcta  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
